.circleci/config.yml

```diff
@@ -1,141 +1,66 @@
-
-commands:
-
-  restore_m2_cache:
-    description: "Restore m2 cache for specified maven task"
-    parameters:
-      cache_name:
-        default: ""
-        type: enum
-        enum: ["", "security-proxy-request-mapper"]
-    steps:
-      - restore_cache:
-          keys:
-            - m2-<<parameters.cache_name>>-{{ checksum "pom.xml" }}
-            - m2-<<parameters.cache_name>>- # used if checksum fails
-            - m2-
-
-  save_m2_cache:
-    description: "Save m2 cache for specified maven task"
-    parameters:
-      cache_name:
-        default: ""
-        type: enum
-        enum: ["", "security-proxy-request-mapper"]
-    steps:
-      - save_cache:
-          key: m2-<<parameters.cache_name>>-{{ checksum "pom.xml" }}
-          paths:
-            - ~/.m2
+version: 2
 
 jobs:
-
   #########################
   # BUILD
   #########################
   build:
+    working_directory: ~/repo
     docker:
       - image: circleci/openjdk:8-jdk
-    working_directory: ~/repo
+        environment:
+          MAVEN_CLI_OPTS: --threads 1C --batch-mode --errors --fail-at-end --show-version -DinstallAtEnd=true -DdeployAtEnd=true -s .circleci/mavenSettings.xml
+          MAVEN_OPTS: -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=WARN -Dorg.slf4j.simpleLogger.showDateTime=true
+
     steps:
-      - setup_remote_docker:
-          docker_layer_caching: true
       - checkout
-      - restore_cache: *restore_m2_cache
+      - restore_cache:
+          key: example-security-proxy-request-mapper-{{ checksum "pom.xml" }}
       - run:
-          name: Build and Unit test
-          command: mvn -s .circleci/mavenSettings.xml clean install -U
+          name: Collect and save dependencies
+          command: mvn $MAVEN_CLI_OPTS dependency:go-offline
       - save_cache:
-          key: m2-{{ checksum "pom.xml" }}
           paths:
             - ~/.m2
-      - persist_to_workspace:
-          root: ~/
-          paths:
-            - repo
+          key: example-security-proxy-request-mapper-{{ checksum "pom.xml" }}
+      - run:
+          name: Build and run unit tests
+          command: mvn $MAVEN_CLI_OPTS package
 
   #########################
   # DOCKERS
   #########################
-  deploy-docker-security-proxy-request-mapper-snapshot:
+  deploy-docker-snapshot:
     docker:
       - image: circleci/openjdk:8-jdk
     working_directory: ~/repo
     steps:
-      - setup_remote_docker:
-          docker_layer_caching: true
       - attach_workspace:
           at: ~/
-      - restore_m2_cache:
-          cache_name: "security-proxy-request-mapper"
+      - restore_cache:
+          key: example-security-proxy-request-mapper-{{ checksum "pom.xml" }}
       - run:
           name: Build snapshot docker artifacts
-          command: |
-            mvn -s .circleci/mavenSettings.xml docker:build -Psnapshot -U
+          command: mvn $MAVEN_CLI_OPTS docker:build -Psnapshot -U
```

Comment:
I don't think the `MAVEN_CLI_OPTS` variable also available in this step without explicitly defining it again. On second thought it might be better to add this variable to the `dev-context` since we'll probably use it in all our Java based projects.

Reply:
After discussion we decided to move those env variables to dev-context on circleci